package.json

```diff
@@ -97,9 +97,19 @@
     "@types/react": "^17.0.0",
     "@types/react-dom": "^17.0.0",
     "@types/react-modal": "^3.10.6",
+    "@typescript-eslint/eslint-plugin": "^4.10.0",
+    "@typescript-eslint/parser": "^4.10.0",
+    "babel-eslint": "^10.1.0",
+    "eslint": "^7.15.0",
```

Comment:
Most of these were in the yarn.lock, but it was complaining when they're referenced without a hard dependency

Reply:
That's because `react-scripts` is taking care of all of these imports. 
For example, `eslint-plugin-react-hooks` isn't mentioned in "our" code, is it. I think it gets dragged in by `create-react-app`. 
Now that we have them in two places, will it trigger weird version mismatches when there's a new `webpack` (for example) that hasn't been tested by `react-scripts`?

Reply:
> but it was complaining when they're referenced without a hard dependency

That's a rather strange problem. And all of this is making me nervous enough to ask you to not do this too please. 

For example, through `react-scripts` (which is what you get when you use `create-react-app`) it has its own set of dependencies (`jest`, `eslint`, `webpack` to name a few). We use and rely on `jest` for running the tests against our Node code. But as you can see, we don't have `jest` listed in `package.json`. 

The only reason you didn't also do `jest` was because there's nowhere where we use `require('jest')` I'm guessing. 

Here's an example of `react-scripts` freaking out if you add a package, with wrong range, that it expects to install by itself. https://github.com/mdn/yari/issues/1659

All in all, those problems are just problems. I don't know what the perfect course of action is. Try to nail the right version of `jest` (for example) to match exactly what `react-scripts` desires. Or to allow `react-scripts` handle the installation of all of these. 

For progress, I think I'd like to go back to the way it was. I.e. *not* let the `eslint` rule that complains about dependencies not written down in package.json.